varnish-http-purge.php

```diff
@@ -347,94 +380,131 @@ public function purgeNoID( $postId ) {
 	 */
 	public function purgePost( $postId ) {
 		
+		global $wp_version;
+		
 		// If this is a valid post we want to purge the post, 
 		// the home page and any associated tags and categories
-
-		$validPostStatus = array("publish", "trash");
-		$thisPostStatus  = get_post_status($postId);
+		$valid_post_status = array( 'publish', 'private', 'trash' );
+		$this_post_status  = get_post_status( $postId );
+
+		// Not all post types are created equal
+		$invalid_post_type = array( 'nav_menu_item', 'revision' );
+		$noarchive_post_type = array( 'post', 'page' );
+		$this_post_type = get_post_type( $postId );
+				
+		// Determine the route for the rest API
+		// This will need to be revisted if WP updates the version.
+		// Future me: Consider an array? 4.7-4.7.3 use v2, and then adapt from there?
+		$rest_api_route  = 'wp/v2'; 
 
 		// array to collect all our URLs
 		$listofurls = array();
 
-		if( get_permalink($postId) == true && in_array($thisPostStatus, $validPostStatus) ) {
-			// If this is a post with a permalink AND it's published or trashed, 
-			// we're going to add a ton of things to flush.
+		// Verify we have a permalink and that we're a valid post status and a not an invalid post type
+		if( get_permalink( $postId ) == true && in_array( $this_post_status, $valid_post_status ) && !in_array( $this_post_type, $invalid_post_type ) ) {
+
+			// Post URL
+			array_push( $listofurls, get_permalink( $postId ) );
+
+			// JSON API Permalink for the post based on type
+			// We only want to do this if the rest_base exists
+			// But we apparently have to force it for posts and pages (seriously?)
+			$post_type_object = get_post_type_object( $postId );	
+			if ( isset( $post_type_object->rest_base ) ) {
+				$rest_permalink = get_rest_url().$rest_api_route.'/'.$post_type_object->rest_base.'/'.$postId.'/';
```

Comment:
Right it works, but if you take a look at https://wordpress.org/news/wp-json/wp/v2/posts and the `_links` property you'll notice that no trailing slash is appended by default.